### PR TITLE
Shell auto-completion for nef

### LINKED ISSUE #136: Add autocompletation to terminal
`nef` has a lot of `commands` and `options` I think is the time to integrate an autocompletion in the installer. So your terminal helps you to choose between all the options.

It should be nice if this autocompletion works on:
- bash
- [zsh](https://ohmyz.sh/)

### PR DESCRIPTION
**Pull request by miguelangel-dev**
## Issues
- Closes: #136 

## Details
Yesterday `swift-argument-parser` released version `0.2.1` adding support to [shell-completion](https://github.com/apple/swift-argument-parser/issues/1). Taking advantage of this new feature, I have upgrade code to this version and modify `Makefile` to install completions scripts after each install.

## Limitations
Right now, it is only compatible with `bash` and `zsh`, but they also have in the roadmap integrate [fish](https://github.com/apple/swift-argument-parser/issues/214)

### CODE REVIEW

File: project/UI/Carbon/Carbon.swift [hunk 1]
```diff
@@ -28,16 +28,16 @@ public struct CarbonCommand: ParsableCommand {
...
     
     @ArgumentParser.Option(help: "Carbon font type")
     private var font: CarbonStyle.Font = .firaCode
     
-    @ArgumentParser.Option(name: .customLong("show-lines"), help: "Shows/hides lines of code [true | false]")
+    @ArgumentParser.Option(name: .customLong("show-lines"), help: "Shows/hides lines of code")
     private var lines: Bool = true
     
-    @ArgumentParser.Option(name: .customLong("show-watermark"), help: "Shows/hides the watermark [true | false]")
+    @ArgumentParser.Option(name: .customLong("show-watermark"), help: "Shows/hides the watermark")
```
Comment (truizlop) [inline project/UI/Carbon/Carbon.swift]: These two can be replaced by `@Flag(default: true, inversion: .prefixedNo)`, allowing us to invoke it with: `--show-lines` or `--no-show-lines`.

Alternatively, we could use `@Flag(default: true, inversion: .prefixedEnableDisable)` to invoke with `--enable-show-lines` or `--disable-show-lines`.

File: project/UI/CarbonPage/CarbonPage.swift [hunk 2]
```diff
@@ -28,16 +28,16 @@ public struct CarbonPageCommand: ParsableCommand {
...
     
     @ArgumentParser.Option(help: "Carbon font type")
     private var font: CarbonStyle.Font = .firaCode
     
-    @ArgumentParser.Option(name: .customLong("show-lines"), help: "Shows/hides lines of code [true | false]")
+    @ArgumentParser.Option(name: .customLong("show-lines"), help: "Shows/hides lines of code")
     private var lines: Bool = true
     
-    @ArgumentParser.Option(name: .customLong("show-watermark"), help: "Shows/hides the watermark [true | false]")
+    @ArgumentParser.Option(name: .customLong("show-watermark"), help: "Shows/hides the watermark")
```
Comment (truizlop) [inline project/UI/CarbonPage/CarbonPage.swift]: These arguments and some others above are the same that other commands are using. Consider extracting them into a structure that conforms to `ParsableArguments` to be able to reuse them.

File: project/UI/Carbon/Carbon.swift [hunk 3]
```diff
@@ -28,16 +28,16 @@ public struct CarbonCommand: ParsableCommand {
     @ArgumentParser.Option(help: "Carbon theme")
     private var theme: CarbonStyle.Theme = .dracula
     
-    @ArgumentParser.Option(help: "export file size [1-5]")
+    @ArgumentParser.Option(help: "export file size")
```
Comment (truizlop) [inline project/UI/Carbon/Carbon.swift]:
```suggestion
    @ArgumentParser.Option(help: "Export file size")
```

File: project/UI/CarbonPage/CarbonPage.swift [hunk 4]
```diff
@@ -28,16 +28,16 @@ public struct CarbonPageCommand: ParsableCommand {
     @ArgumentParser.Option(help: "Carbon theme")
     private var theme: CarbonStyle.Theme = .dracula
     
-    @ArgumentParser.Option(help: "export file size [1-5]")
+    @ArgumentParser.Option(help: "export file size")
```
Comment (truizlop) [inline project/UI/CarbonPage/CarbonPage.swift]:
```suggestion
    @ArgumentParser.Option(help: "Export file size")
```

**Review by truizlop — CHANGES_REQUESTED** [review]

File: project/UI/CarbonPage/CarbonPage.swift (hunk 2 above)
Comment (miguelangel-dev) [reply project/UI/CarbonPage/CarbonPage.swift]: I will review it in a future PR, I don't want to add more noise to this one.

**Review by truizlop — APPROVED** [review]

**Review by truizlop — APPROVED** [review]

**Pull request closed without merging**